framework/graphics/fps_info.cpp

```diff
@@ -90,19 +102,71 @@ void FpsInfo::BeginFrame(uint64_t frame)
         {
             measurement_start_time_ = util::datetime::GetTimestamp();
             started_measurement_    = true;
+            frame_durations_.clear();
         }
     }
+
+    frame_start_time_ = util::datetime::GetTimestamp();
 }
 
 void FpsInfo::EndFrame(uint64_t frame)
 {
     if (started_measurement_ && !ended_measurement_)
     {
+        frame_durations_.push_back(util::datetime::DiffTimestamps(frame_start_time_, util::datetime::GetTimestamp()));
+
         // Measurement frame range end is non-inclusive, as opposed to trim frame range
         if (frame >= measurement_end_frame_ - 1)
         {
             measurement_end_time_ = util::datetime::GetTimestamp();
             ended_measurement_    = true;
+
+            // Save measurements to file
+            if (!measurement_file_name_.empty())
+            {
+                double   start_time   = util::datetime::ConvertTimestampToSeconds(measurement_start_time_);
+                double   end_time     = util::datetime::ConvertTimestampToSeconds(measurement_end_time_);
+                double   diff_time    = GetElapsedSeconds(measurement_start_time_, measurement_end_time_);
+                uint64_t total_frames = measurement_end_frame_ - measurement_start_frame_;
+                double   fps          = static_cast<double>(total_frames) / diff_time;
+
+                nlohmann::json file_content = { { "frame_range",
```

Comment:
Your previous version had an object, something like `"application":{ "capture_name": "a/file/name"}` in here too. Just checking the deletion is deliberate (it was added to the tree in the manager previously).

Reply:
Yes, it was deliberate, this is not needed, was wrongly implemented, and would be a bit hard/annoying to implement here. Not worth it.